Adafruit_MCP23017.cpp

```diff
@@ -137,6 +137,45 @@ void Adafruit_MCP23017::begin(uint8_t addr, TwoWire *theWire) {
  */
 void Adafruit_MCP23017::begin(TwoWire *theWire) { begin(0, theWire); }
 
+/**
+ * Changes the adress to which MCP23017 is talked to.
+ * Initializing all ports as input is optional.
+ * This enables the option to control multiple MCP23017 with one instance.
+ * Both adress syntax (i2x) = 0x20 or only the configurable part is supported.
+ * @param addr Selected address
+ * @param initAsInput If true initializing all ports as input, otherwise they
+ * stay as they are
+ */
+void Adafruit_MCP23017::changeAddress(uint8_t addr, bool initAsInput) {
+  // allow using both syntaxes, complete i2x address or address pin setting
+  if ((addr >= MCP23017_ADDRESS) && (addr < (MCP23017_ADDRESS + 8))) {
+    addr -= MCP23017_ADDRESS;
+  }
+
+  if (addr > 7) {
+    addr = 7;
+  }
+
+  i2caddr = addr;
+
+  if (initAsInput) {
+    // set defaults only if wanted.
+    // all inputs on port A and B
+    writeRegister(MCP23017_IODIRA, 0xff);
+    writeRegister(MCP23017_IODIRB, 0xff);
+  }
+}
+
+/**
```

Comment:
Hi @Steinies 
I also suggest to remove this and simply add a default value to `false `to the `initAsInput `parameter in the header so that `changeAddress(addr) `without second parameter will not reinitialize port mode.
```
void changeAddress(uint8_t addr, bool initAsInput=false);
```